src/internet/radiobrowser/radiobrowserservice.cpp

```diff
@@ -0,0 +1,448 @@
+    QStandardItem* item = new QStandardItem(
+        IconLoader::Load("icon_radio", IconLoader::Lastfm), QString());
+    item->setText(branch.name);
+    item->setData(branch.type, InternetModel::Role_Type);
+    item->setData(branch.listUrl, RadioBrowserService::Role_ListUrl);
+    item->setData(branch.itemsUrl, RadioBrowserService::Role_ItemsUrl);
+    item->setData(true, InternetModel::Role_CanLazyLoad);
+    root_->appendRow(item);
+  }
+}
+
+void RadioBrowserService::RefreshCategory(QStandardItem* item) {
+  if (!EnsureServerConfig()) {
+    ShowConfig();
+    return;
+  }
+
+  QString determinedUrl = item->data(RadioBrowserService::Role_ListUrl)
+                              .toString()
+                              .arg(main_server_url_);
+  QUrl url(determinedUrl);
+
+  QNetworkReply* reply = network_->get(QNetworkRequest(url));
+  int task_id = app_->task_manager()->StartTask(tr("Getting channels"));
+
+  connect(reply, &QNetworkReply::finished,
+          [=] { this->RefreshCategoryFinished(reply, task_id, item); });
+}
+
+void RadioBrowserService::RefreshCategoryItem(QStandardItem* item) {
+  if (!EnsureServerConfig()) {
+    ShowConfig();
+    return;
+  }
+
+  QStandardItem* parent = item->parent();
+  QString determinedUrl = parent->data(RadioBrowserService::Role_ItemsUrl)
+                              .toString()
+                              .arg(main_server_url_, item->text());
+  QUrl url(determinedUrl);
+
+  QNetworkReply* reply = network_->get(QNetworkRequest(url));
+  int task_id = app_->task_manager()->StartTask(tr("Getting channels"));
+
+  connect(reply, &QNetworkReply::finished,
+          [=] { this->RefreshStreamsFinished(reply, task_id, item); });
+}
+
+void RadioBrowserService::RefreshTop100(QStandardItem* item) {
+  if (!EnsureServerConfig()) {
+    ShowConfig();
+    return;
+  }
+
+  QString determinedUrl = item->data(RadioBrowserService::Role_ItemsUrl)
+                              .toString()
+                              .arg(main_server_url_);
+  QUrl url(determinedUrl);
+
+  QNetworkReply* reply = network_->get(QNetworkRequest(url));
+  int task_id = app_->task_manager()->StartTask(tr("Getting channels"));
+
+  connect(reply, &QNetworkReply::finished,
+          [=] { this->RefreshStreamsFinished(reply, task_id, item); });
+}
+
+void RadioBrowserService::ShowContextMenu(const QPoint& global_pos) {
+  if (!model()->current_index().isValid()) return;
+  QStandardItem* item = model()->itemFromIndex(model()->current_index());
+  if (!item) return;
+
+  switch (item->data(InternetModel::Role_Type).toInt()) {
+    case InternetModel::Type_Service:
+      GetContextMenu(item)->popup(global_pos);
+      break;
+    case InternetModel::Type_Track:
+      GetStationMenu(item)->popup(global_pos);
+      break;
+    default:
+      break;
+  }
+}
+
+QMenu* RadioBrowserService::GetContextMenu(QStandardItem* item) {
+  Q_UNUSED(item);
+  if (!context_menu_) {
+    context_menu_ = new QMenu;
```

Comment:
This is leaked. You should change this to use a `unique_ptr` or better yet, use `Lazy<QMenu>` if possible.

Reply:
I was planning on moving conext_menu_ to the base class in a future pull request, so if you do use Lazy<>, try to follow the pattern here:
https://github.com/clementine-player/Clementine/blob/master/src/internet/core/cloudfileservice.cpp#L111